armonik/variables.tf

```diff
@@ -325,71 +263,80 @@ variable "authentication" {
   }
 }
 
-# The name of the secrets.
-variable "fluent_bit_secret_name" {
-  description = "the name of the fluent-bit secret"
-  type        = string
-  default     = "fluent-bit"
-}
-
-variable "grafana_secret_name" {
-  description = "the name of the grafana secret"
-  type        = string
-  default     = "grafana"
-}
-
-variable "prometheus_secret_name" {
-  description = "the name of the prometheus secret"
-  type        = string
-  default     = "prometheus"
-}
-
-variable "metrics_exporter_secret_name" {
-  description = "the name of the metrics exporter secret"
-  type        = string
-  default     = "metrics-exporter"
-}
-
-variable "partition_metrics_exporter_secret_name" {
-  description = "the name of the partition metrics exporter secret"
-  type        = string
-  default     = "partition-metrics-exporter"
+# The output of modules.
+variable "fluent_bit_output" {
+  description = "the fluent-bit module output"
+  type = object({
+    configmaps = object({
+      envvars = string
+      config  = string
+    })
+    container_name = string
+    image          = string
+    is_daemonset   = bool
+    tag            = string
+  })
+  default = null
 }
 
-variable "seq_secret_name" {
-  description = "the name of the seq secret"
-  type        = string
-  default     = "seq"
+variable "grafana_output" {
+  description = "the grafana module output"
+  type = object({
+    host = string
+    port = string
+    url  = string
+  })
+  default = null
 }
 
-variable "shared_storage_secret_name" {
-  description = "the name of the shared-storage secret"
-  type        = string
-  default     = "shared-storage"
-}
 
-variable "deployed_object_storage_secret_name" {
-  description = "the name of the deployed-object-storage secret"
-  type        = string
-  default     = "deployed-object-storage"
+variable "prometheus_output" {
+  description = "the prometheus module output"
+  type = object({
+    host = string
+    port = string
+    url  = string
+  })
+  default = null
 }
 
-variable "deployed_table_storage_secret_name" {
-  description = "the name of the deployed-table-storage secret"
-  type        = string
-  default     = "deployed-table-storage"
+variable "metrics_exporter_output" {
+  description = "the metrics exporter module output"
+  type = object({
+    host      = string
+    name      = string
+    namespace = string
+    port      = string
+    url       = string
+  })
+  default = null
 }
 
-variable "deployed_queue_storage_secret_name" {
-  description = "the name of the deployed-queue-storage secret"
-  type        = string
-  default     = "deployed-queue-storage"
+variable "seq_output" {
+  description = "the seq module output"
+  type = object({
+    host    = string
+    port    = string
+    url     = string
+    web_url = string
+  })
+  default = null
 }
 
-variable "s3_secret_name" {
-  description = "the name of the S3 secret"
-  type        = string
-  default     = "s3"
+variable "shared_storage_settings" {
```

Comment:
I think we could completely implement it using the new configuration mechanism.

Reply:
Some of the information are used in the ingress-config map not as env or mount